translator/translator/translator.py

```diff
@@ -0,0 +1,217 @@
+    return rx.box(
+        rx.heading("Translator 🗺", size="8"),
+        rx.text(
+            "Translate things and post them as messages!",
+            color_scheme="gray",
+        ),
+        margin_bottom="1rem",
+    )
+
+
+def down_arrow() -> rx.Component:
+    return rx.vstack(
+        rx.icon(
+            tag="arrow_down",
+            color="gray",
+        ),
+    )
+
+
+def text_box(text) -> rx.Component:
+    return rx.text(
+        text,
+        background_color="white",
+        padding="1rem",
+        border_radius="8px",
+    )
+
+
+def past_translation(message: Message) -> rx.Component:
+    """A layout that contains a past translation."""
+    return rx.box(
+        rx.flex(
+            text_box(message.original_text),
+            down_arrow(),
+            text_box(message.text),
+            rx.hstack(
+                rx.text(message.to_lang),
+                rx.text("·"),
+                rx.text(message.created_at),
+                spacing="2",
+                font_size="0.8rem",
+                color=rx.color("gray", 11),
+            ),
+            spacing="2",
+            direction="column",
+        ),
+        background_color=rx.color("gray", 3),
+        padding="1rem",
+        border_radius="8px",
+    )
+
+
+def smallcaps(text: str, **props):
+    """A smallcaps text component."""
+    return rx.text(
+        text,
+        font_size="0.7rem",
+        font_weight="bold",
+        text_transform="uppercase",
+        letter_spacing="0.05rem",
+        **props,
+    )
+
+
+def output():
+    """The output layout that contains the translated text."""
+    return rx.stack(
+        smallcaps(
+            "Output",
+            color=rx.color("gray", 9),
+            background_color="white",
+            position="absolute",
+            top="-0.5rem",
+        ),
+        rx.text(State.output, size="4"),
+        padding="1rem",
+        border="1px solid",
+        border_color=rx.color("gray", 7),
+        border_radius="8px",
+        position="relative",
+    )
+
+
+def translation_form() -> rx.Component:
+    """This is a form that contains:
+    - Text input: what to translate
+    - Select: the destination language
+    - Output: the translation
+    - Submit button: to post the text input as part of the past translations view
+    """
+    return rx.form(
+        rx.flex(
+            rx.input(
+                placeholder="Text to translate",
+                value=State.text,
+                on_change=State.set_text,
+                debounce_timeout=300,
+                size="3",
+                name="text",
+            ),
+            rx.select(
+                placeholder="Select a language",
+                items=LANGUAGE_NAMES,
+                value=State.lang,
+                on_change=State.set_lang,
+                margin_top="1rem",
+                size="3",
+                width="100%",
+            ),
+            output(),
+            rx.form.submit(
```

Comment:
Should we use `rx.form.submit` or use `rx.button(type="submit")` which is the current way?

Reply:
actually `rx.button` defaults to a submit button now; since in html that is the default.

Reply:
@picklelo think you have a question about that

Reply:
Do all buttons default bo submit? What if there's a use case where they want another button in their form?

Reply:
Okay seems this is the default behavior now, we can keep it